### PR TITLE
Refactor | V1-240 - Refactored Routes structure. Deleted layout in components

### PR DESCRIPTION
**Pull request by ostapdans99**
- Frontend 

1. Refactored Routes structure.
2. Added new component for PageTitles.
3. Deleted layouts in components.
4. Added navigation state for loign/logout

### CODE REVIEW

File: frontend/src/pages/CoursesList/CoursesList.tsx [hunk 1]
```diff
@@ -29,52 +30,47 @@ const CoursesList: FC<ICourseProps> = ({
   lastCourseRef,
   isAdmin,
 }) => (
-  <AuthorizedLayout pageName="Courses List">
+  <PageTitle title="Courses List">
```
Comment (irina-sinitsyna) [inline frontend/src/pages/CoursesList/CoursesList.tsx:33]: let's reuse title from constant?

File: frontend/src/pages/NotFound/NotFoundContainer.tsx [hunk 2]
```diff
@@ -1,25 +1,13 @@
...
-        <NotFound />
-      </DefaultWrapper>
-    </DefaultLayout>
-  );
-};
+const NotFoundContainer: FC = () => (
+  <PageTitle title="Not found">
+    <NotFound />
+  </PageTitle>
+);
```
Comment (irina-sinitsyna) [inline frontend/src/pages/NotFound/NotFoundContainer.tsx:11]: do we still need container in this case?

File: frontend/src/pages/Profile/UserSkills/UserSkills.tsx [hunk 3]
```diff
@@ -42,62 +40,60 @@ const UserSkills: React.FC<Props> = ({
...
+        value={searchSkill}
+      />
+      <Divider />
+    </SearchWrapper>
+    <SkillsList>
+      {technologies && technologies.length ? (
+        technologies.map((techGroup, id) => {
+          const isDividerVisible = id < technologies.length - 1;
+          return (
+            <div key={`${techGroup.group.name}_id`}>
```
Comment (irina-sinitsyna) [inline frontend/src/pages/Profile/UserSkills/UserSkills.tsx:73]: as far as I understood, the group name is unique, so we need to add an `_id` as plain text?

```suggestion
            <div key={techGroup.group.name}> 
```

File: frontend/src/pages/Profile/UserSkills/UserSkills.tsx [hunk 4]
```diff
@@ -42,62 +40,60 @@ const UserSkills: React.FC<Props> = ({
...
+              <SkillsListItem>
+                <SkillTitle>
+                  {techGroup.isPrimary && <Star alt="primary" src={starContained} />}
+                  <Title>{techGroup.group.name || 'No group'}</Title>
+                </SkillTitle>
+                <SkillsInfoList>
+                  {techGroup.achievedSkills.map((skillInfo) => (
+                    <React.Fragment key={`${skillInfo.skill.name}_id`}>
```
Comment (irina-sinitsyna) [inline frontend/src/pages/Profile/UserSkills/UserSkills.tsx:81]: the same

File: frontend/src/pages/Profile/UserSkills/UserSkills.tsx (hunk 3 above)
Comment (ostapdans99) [reply frontend/src/pages/Profile/UserSkills/UserSkills.tsx:73]: yep, but it not my code. i just removed Suspense here. Аlso this component should be refаctored.

File: frontend/src/pages/Profile/UserSkills/UserSkills.tsx (hunk 4 above)
Comment (ostapdans99) [reply frontend/src/pages/Profile/UserSkills/UserSkills.tsx:81]: The sаme. Аnd if we аre tаlking аbout mistаkes, аt this cаse dont need to use Frаgment with key.

Comment (irina-sinitsyna) [reply frontend/src/pages/Profile/UserSkills/UserSkills.tsx:81]: agree

File: frontend/src/pages/NotFound/NotFoundContainer.tsx (hunk 2 above)
Comment (ostapdans99) [reply frontend/src/pages/NotFound/NotFoundContainer.tsx:11]: yep, we need. but аlso we саn do this аt NotFound component.

File: frontend/src/pages/CoursesList/CoursesList.tsx (hunk 1 above)
Comment (ostapdans99) [reply frontend/src/pages/CoursesList/CoursesList.tsx:33]: yes, we can, but if we will do this, we shold use this constants at route configs. We can do this inside other task.

File: frontend/src/api/auth/getAuth.tsx [hunk 5]
```diff
@@ -10,15 +10,21 @@ import { COOKIE_VALUES } from 'constants/authConstants';
 import { errorSnackbar, successSnackbar, successSnackbarMessage } from 'constants/snackbarVariant';
 import { ITokenResponse } from 'types/auth';
 
+import { TLocation } from './type';
```
[the remark this answers is not in the document]
Comment (ostapdans99) [reply frontend/src/api/auth/getAuth.tsx:13]: done

File: frontend/src/pages/NotFound/NotFoundContainer.tsx (hunk 2 above)
Comment (ostapdans99) [reply frontend/src/pages/NotFound/NotFoundContainer.tsx:11]: sorry, i understand, done

**Review by irina-sinitsyna — APPROVED** [review]

**Pull request merged**